### ISSUE TITLE
error parsing haml with coffeescript

### ISSUE THREAD

**Issue by Cohen-Carlisle** [presidentbeef/brakeman#1299]
### Background

Brakeman version: 4.3.1
Rails version: 5.1.6.1
Ruby version: 2.4.3

Link to Rails application code: =(

#### Parse Error

Error: "coffee" filter's coffee_script dependency missing: try installing it or adding it to your Gemfile
Location: While compiling HAML in /Users/ccarlisle/code/criteria/app/views/outcome_types/show.html.haml

Minimal example that does not parse:

```haml
.row
  - cache @outcome_type do
    .col-md-4
      = back_link_to account_reasons_path(@account), 'All Reasons', :blue

      .panel.panel-primary.outcomes-tool
        %header.trans.panel-heading
          %h3
            Edit Outcome Type
            %i.fa.fa-pencil
        .panel-body
          = form_for [@outcome_type.account, @outcome_type],
                     html: { class: 'validate' } do |f|
            .form-group
              = f.label :name, class: 'control-label'
              = f.text_field :name, class: 'form-control required'

            .form-group
              = f.label :ncpdp_type, class: 'control-label' do
                NCPDP Type
              = f.text_field :ncpdp_type, class: 'form-control'

            .form-group.color-chooser
              = f.label :color, class: 'control-label'
              %small (Click to open a color picker in Chrome/Firefox)

              %p
                = f.color_field :color, class: 'form-control'

              .form-inline
                %p Or select one of the default colors:

                - OutcomeType::DEFAULT_COLORS.each do |color|
                  .input-group
                    %button.btn.default-color{ data: {color: color},
                                               style: "background-color: #{color};" }

            .form-group
              = f.label :note, class: 'control-label'
              = f.text_area :note, class: 'form-control'

            = f.submit_once class: 'btn btn-primary btn-block update-outcome-type'

  .col-md-8
    .panel.panel-primary.outcomes-tool
      %header.panel-heading
        %h4
          Reasons
          %i.fa.fa-briefcase
          %small= "#{@reasons.total_count} found"
      .panel-body{ style: "overflow: visible;" }
        = form_tag [@account, @outcome_type], method: :get do
          .row.thin.thin-margin
            .col-sm-11
              = search_field_tag :for_name_or_desc, nil,
                                 placeholder: 'Reason Name or Description',
                                 class: 'form-control',
                                 value: @query_params[:for_name_or_desc]
            .col-sm-1
              = button_tag class: 'btn btn-block btn-tool-color btn-trans', id: 'reason-search' do
                %i.fa.fa-search
      %table.table.table-hover#outcome-reasons
        %tbody
          - @reasons.each do |reason|
            %tr
              %td= link_to reason.name, account_reason_path(@outcome_type.account, reason), class: 'edit-outcome-reason'
      %footer.panel-footer
        = render partial: 'pagination',
                 locals: { pageable: @reasons, query_params: @query_params }

- content_for :scripts do
  :coffeescript
    form = $('form[id*=edit_outcome_type]')
    colorInput = $('#outcome_type_color', form)

    form.on 'click', '.default-color', (event) ->
      colorInput.val $(event.target).data('color')
      false
```

### Notes
I've found a few workarounds, but I'm really unsure of which is the best.
1. I can just `gem install coffee-script`
2. If I add brakeman to the Gemfile and `bundle exec brakeman` it works because coffee-script gem is installed that way (but it is installed to $APP_ROOT/vendor/bundle/, so unavailable globally)
3. If I change the extension of the file to .hamlc, it works. I'm not personally familiar with haml vs hamlc, however.

**Comment by presidentbeef** [comment]
Hi Cohen,

Thank you for bringing up this issue.

Your workarounds (1 or 2) are the correct approach. Brakeman does not include dependencies to handle HAML filters, but they should work fine if you make them available. How you do that is up to you and what works best for your environment.

Workaround (3) just causes Brakeman to not see the file at all, which is probably not what you want.

**Comment by Cohen-Carlisle** [comment]
Thanks! Feel free to close if this is expected behavior (as it seems it is).

**Comment by presidentbeef** [comment]
👍

**Comment by presidentbeef** [comment]
This is actually addressed in #1314. Any additional filters/embedded languages that need to be supported in the future can probably follow the same model without requiring new dependencies or for users to have undeclared dependencies installed.
